File: Src/Intelij IDEA/SDK/Java/Eruru.ChatRobotRPC/src/main/java/org/eruru/chatrobotrpc/enums/ChatRobotEventType.java
package org.eruru.chatrobotrpc.enums;

/**
 * 事件类型
 */
public enum ChatRobotEventType {

	/**
	 * 收到自身消息
	 */
	receivedOwnMessage (2099),
	/**
	 * 群公告改变
	 */
	groupAnnouncementChanged (2013),
	/**
	 * 好友签名改变
	 */
	friendSignatureChanged (1004),
	/**
	 * 说说被评论
	 */
	talkWasCommented (1005),
	/**
	 * 好友正在输入
	 */
	friendIsTyping (1006),
	/**
	 * 好友今天首次发起会话
	 */
	friendFirstChatToday (1007),
	/**
	 * 被好友抖动
	 */
	wasJitterByFriend (1008),
	/**
	 * 收到财付通转账
	 */
	receivedTenPayTransfer (80001),
	/**
	 * 添加了新账号
	 */
	addedNewAccount (11000),
	/**
	 * QQ登录完成
	 */
	qqloggedIn (11001),
	/**
	 * QQ被手动离线
	 */
	qqWasOfflineByManual (11002),
	/**
	 * QQ被强制离线
	 */
	qqWasOfflineByForce (11003),
	/**
	 * QQ长时间无响应或掉线
	 */
	qqNoResponseForLongTimeOrOffline (11004);

	private final int value;

	ChatRobotEventType (int value) {
		this.value = value;
	}

	/**
	 * 获取枚举对应的整形
	 */
	public int getValue () {
		return value;
	}

	/**
	 * 通过整形获取对应的枚举
	 */
	public static ChatRobotEventType get (int value) {
		for (ChatRobotEventType item : ChatRobotEventType.values ()) {
			if (item.value == value) {
				return item;
			}
		}
		return null;
	}

}

File: Src/Visual Studio/SDK/C#/Eruru.ChatRobotRPC For Shared Project/Enums/ChatRobotEventType.cs
namespace Eruru.ChatRobotRPC {

	/// <summary>
	/// 消息类型
	/// </summary>
	public enum ChatRobotEventType {

		/// <summary>
		/// 收到自身消息
		/// </summary>
		ReceivedOwnMessage = 2099,
		/// <summary>
		/// 群公告改变
		/// </summary>
		GroupAnnouncementChanged = 2013,
		/// <summary>
		/// 好友签名改变
		/// </summary>
		FriendSignatureChanged = 1004,
		/// <summary>
		/// 说说被评论
		/// </summary>
		TalkWasCommented = 1005,
		/// <summary>
		/// 好友正在输入
		/// </summary>
		FriendIsTyping = 1006,
		/// <summary>
		/// 好友今天首次发起会话
		/// </summary>
		FriendFirstChatToday = 1007,
		/// <summary>
		/// 被好友抖动
		/// </summary>
		WasJitterByFriend = 1008,
		/// <summary>
		/// 收到财付通转账
		/// </summary>
		ReceivedTenPayTransfer = 80001,
		/// <summary>
		/// 添加了新账号
		/// </summary>
		AddedNewAccount = 11000,
		/// <summary>
		/// QQ登录完成
		/// </summary>
		QQloggedIn = 11001,
		/// <summary>
		/// QQ被手动离线
		/// </summary>
		QQWasOfflineByManual = 11002,
		/// <summary>
		/// QQ被强制离线
		/// </summary>
		QQWasOfflineByForce = 11003,
		/// <summary>
		/// QQ长时间无响应或掉线
		/// </summary>
		QQNoResponseForLongTimeOrOffline = 11004

	}

}